deeposlandia/datasets/tanzania.py

```diff
@@ -150,29 +193,121 @@ def _preprocess(self, image_filename, output_dir, labelling):
         raw_img_width = raster.RasterXSize
         raw_img_height = raster.RasterYSize
         result_dicts = []
-        logger.info("Raw image size: %s, %s" % (raw_img_width, raw_img_height))
         logger.info("Image filename: %s" % image_filename)
-
-        labels = None
-        if labelling:
-            label_filename = (image_filename
-                              .replace("images", "labels")
-                              .replace(".tif", ".geojson"))
-            labels = gpd.read_file(label_filename)
-            labels = labels.loc[~labels.geometry.isna(), ["condition", "geometry"]]
-            none_mask = [lc is None for lc in labels.condition]
-            labels.loc[none_mask, "condition"] = "Complete"
+        logger.info("Raw image size: %s, %s" % (raw_img_width, raw_img_height))
 
         for x in range(0, raw_img_width, self.image_size):
             for y in range(0, raw_img_height, self.image_size):
                 tile_results = self._preprocess_tile(x, y, image_filename,
-                                                     output_dir,
-                                                     raster, labels)
+                                                     output_dir, raster)
                 result_dicts.append(tile_results)
         del raster
         return result_dicts
 
 
+    def _preprocess_for_training(self, image_filename, output_dir, nb_images):
+        """Resize/crop then save the training & label images
+
+        Parameters
+        ----------
+        image_filename : str
+            Full path towards the image on the disk
+        output_dir : str
+            Output path where preprocessed image must be saved
+
+        Returns
+        -------
+        dict
+            Key/values with the filenames and label ids
+        """
+        raster = gdal.Open(image_filename)
+        raw_img_width = raster.RasterXSize
+        raw_img_height = raster.RasterYSize
+        image_data = raster.ReadAsArray()
+        image_data = np.swapaxes(image_data, 0, 2)
+        result_dicts = []
+        logger.info("Image filename: %s" % image_filename)
+        logger.info("Raw image size: %s, %s" % (raw_img_width, raw_img_height))
+
+        label_filename = (image_filename
+                          .replace("images", "labels")
+                          .replace(".tif", ".geojson"))
+        labels = gpd.read_file(label_filename)
+        labels = labels.loc[~labels.geometry.isna(), ["condition", "geometry"]]
+        none_mask = [lc is None for lc in labels.condition]
```

Comment:
Can `labels.condition.isnull()` work?

Reply:
Seems to be OK with this alternative function as well. Anyway I keep `labels.condition.notnull()`.